include/blade/modules/guppi/reader.hh

```diff
@@ -76,6 +76,48 @@ class BLADE_API Reader : public Module {
     explicit Reader(const Config& config, const Input& input);
     const Result preprocess(const cudaStream_t& stream = 0) final;
 
+    // Dimension getters
+    const U64 getStepNumberOfAntennas() const {
+        return this->config.stepNumberOfAspects;
+    }
+
+    const U64 getStepNumberOfFrequencyChannels() const {
+        return this->config.stepNumberOfFrequencyChannels;
+    }
+
+    const U64 getStepNumberOfTimeSamples() const {
+        return this->config.stepNumberOfTimeSamples;
+    }
+
+    const U64 getStepNumberOfPolarizations() const {
+        return this->getDatashape()->n_pol;
+    }
+
+    const U64 getStepOutputBufferSize() {
+        return this->getStepOutputBufferDims().size();
+    }
+
+    const U64 getTotalNumberOfAntennas() const {
+        return this->getDatashape()->n_aspect;
+    }
+
+    const U64 getTotalNumberOfFrequencyChannels() const {
+        return this->getDatashape()->n_aspectchan;
+    }
+
+    const U64 getTotalNumberOfTimeSamples() const {
+        return this->getDatashape()->n_time * this->gr_iterate.n_block;
+    }
+
+    const U64 getTotalNumberOfPolarizations() const {
+        return this->getDatashape()->n_pol;
+    }
+
+    const U64 getTotalOutputBufferSize() {
+        return this->getStepOutputBufferDims().size();
+    }
+
+
```

Comment:
Why one method for each dimension value as opposed to `getTotalOutputBufferDims` and `getStepOutputBufferDims`? This breaks the "convention" used by other modules (one method gets'em all).

Reply:
Will clean up.